packages/state-transition/test/perf/block/processWithdrawals.test.ts

```diff
@@ -0,0 +1,107 @@
+import {itBench} from "@dapplion/benchmark";
+import {ssz} from "@lodestar/types";
+import {createIChainForkConfig} from "@lodestar/config";
+import {BLS_WITHDRAWAL_PREFIX, ETH1_ADDRESS_WITHDRAWAL_PREFIX} from "@lodestar/params";
+
+import {CachedBeaconStateCapella} from "../../../src/index.js";
+import {getExpectedWithdrawals} from "../../../src/block/processWithdrawals.js";
+import {numValidators} from "../util.js";
+import {createCachedBeaconStateTest} from "../../utils/state.js";
+
+// PERF: Fixed cost for MAX_WITHDRAWALS_PER_PAYLOAD probes
+//  + cost 'proportional' to $VALIDATOR_COUNT with balances under MAX_EFFECTIVE_BALANCE or
+//    having BLS withdrawal credential prefix as that validator probe is wasted.
+//
+// Best case:
+//  All Validator have balances > MAX_EFFECTIVE_BALANCE and ETH1 withdrawal credential prefix set
+//
+// Worst case:
+//  All balances are low enough or withdrawal credential not set
+
+describe("capella getExpectedWithdrawals", () => {
+  const vc = numValidators;
+  const testCases: {id: string; lowBalanceRatio: number; blsCredentialRatio: number}[] = [
+    // Best case when every probe results into a withdrawal candidate
+    {id: "bestcase", lowBalanceRatio: 0, blsCredentialRatio: 0},
+    // Normal case based on mainnet conditions: mainnet network conditions: 95% reward rate
+    {id: "normalcase - 100% withdrawals enabled", lowBalanceRatio: 0.05, blsCredentialRatio: 0},
+    {id: "normalcase - 70% withdrawals enabled", lowBalanceRatio: 0.05, blsCredentialRatio: 0.3},
+    {id: "normalcase - 40% withdrawals enabled", lowBalanceRatio: 0.05, blsCredentialRatio: 0.6},
+    {id: "normalcase - 10% withdrawals enabled", lowBalanceRatio: 0.05, blsCredentialRatio: 0.9},
+    // Degraded case
+    {id: "slighly degraded case - 90% withdrawals enabled", lowBalanceRatio: 0.1, blsCredentialRatio: 0.1},
+    {id: "degraded case - 90% withdrawals enabled", lowBalanceRatio: 0.3, blsCredentialRatio: 0.1},
+    {id: "highly degraded case - 90% withdrawals enabled", lowBalanceRatio: 0.4, blsCredentialRatio: 0.1},
+    // Worst case: All effective balance are updated
+    // NOTE: The maximum bad case will practically never happen and it's too slow.
```

Comment:
What's the maximum bad case and what do you mean "it's too slow"?

Reply:
sorry its a copy overr from the comment of the file i used as a template to prepare test, will cleanup